File: ClickCounter/src/main/java/edu/luc/etl/cs313/android/clickcounter/BoundedCounterWrapper.java
package edu.luc.etl.cs313.android.clickcounter;

import edu.luc.etl.cs313.misc.boundedcounter.cli.BoundedCounter;

/**
 * Adapter for using the CLI bounded counter as the Android click counter model.
 */
public class BoundedCounterWrapper implements ClickCounterModel {

    final BoundedCounter counter;

    public BoundedCounterWrapper(final BoundedCounter counter) {
        this.counter = counter;
    }

    @Override
    public void increment() {
        counter.increment();
    }

    @Override
    public void decrement() {
        counter.decrement();
    }

    @Override
    public void reset() {
        while (!counter.isEmpty()) {
            counter.decrement();
        }
    }

    @Override
    public int get() {
        return counter.get();
    }

    @Override
    public boolean isFull() {
        return counter.isFull();
    }

    @Override
    public boolean isEmpty() {
        return counter.isEmpty();
    }
}

File: app/src/main/kotlin/edu/luc/etl/cs313/android/clickcounter/BoundedCounterWrapper.kt
package edu.luc.etl.cs313.android.clickcounter

import edu.luc.etl.cs313.misc.boundedcounter.cli.BoundedCounter

/** Adapter for using the CLI bounded counter as the Android click counter model.  */
class BoundedCounterWrapper(val counter: BoundedCounter) : ClickCounterModel {

    override fun increment() {
        counter.increment()
    }

    override fun decrement() {
        counter.decrement()
    }

    override fun reset() {
        while (!counter.isEmpty) {
            counter.decrement()
        }
    }

    override fun get(): Int {
        return counter.get()
    }

    override val isFull: Boolean
        get() = counter.isFull

    override val isEmpty: Boolean
        get() = counter.isEmpty
}
